pyanaconda/modules/boss/installation.py

```diff
@@ -158,29 +153,35 @@ def _relabel_files(self):
         """
         execWithRedirect("restorecon", ["-ir", ANACONDA_LOG_DIR], root=self._sysroot)
 
-    def _copy_file_to_sysroot(self, src, dest):
-        """Copy a file, if it exists.
+    def _copy_file_to_sysroot(self, src, dest, mode=0o0600):
+        """Copy a file, if it exists, and set its access bits.
 
         :param str src: path to source file
         :param str dest: path to destination file within sysroot
+        :param int mode: access bits to set for the file
         """
         if os.path.exists(src):
             log.info("Copying file: %s -> %s", src, dest)
+            full_dest_path = join_paths(self._sysroot, dest)
             shutil.copyfile(
                 src,
-                join_paths(self._sysroot, dest)
+                full_dest_path
             )
+            os.chmod(full_dest_path, mode)
 
-    def _copy_tree_to_sysroot(self, src, dest):
-        """Copy a directory tree, if it exists.
+    def _copy_tree_to_sysroot(self, src, dest, mode=0o0600):
+        """Copy a directory tree, if it exists, and set its access bits.
 
         :param str src: path to source directory
         :param str dest: path to destination directory within sysroot
+        :param int mode: access bits to set for the directory
         """
         if os.path.exists(src):
             log.info("Copying directory tree: %s -> %s", src, dest)
+            full_dest_path = join_paths(self._sysroot, dest)
             shutil.copytree(
                 src,
-                join_paths(self._sysroot, dest),
+                full_dest_path,
                 dirs_exist_ok=True
             )
+            os.chmod(full_dest_path, mode)
```

Comment:
Does it work fine if `chmod` is called only for the top directory?

Reply:
I wondered about this too. But it works exactly as before, because chmod in bash is not recursive by default.